EC-Switch-Seclog.py

```diff
@@ -0,0 +1,973 @@
+
+            response = client_ac.describe_pending_aggregation_requests() # first call may return an empty list of PendingAggregationRequests
+            
+            count = 0
+            while count <= 10:
+                
+                if response['PendingAggregationRequests']:
+                    break
+                time.sleep(5)
+                count=count+1
+                response = client_ac.describe_pending_aggregation_requests()
+                
+
+            if  'PendingAggregationRequests' not in response or not response['PendingAggregationRequests']:
+                print(f"\033[2K\033[1GAssociate AWSConfig linked account to target SECLOG account error: no PendingAggregationRequests found [{Status.FAIL.value}]")
+                print("Exiting...")
+                sys.exit(1)
+
+
+            for pendingAggregationRequests in response['PendingAggregationRequests']:
+                
+                if tseclog_id in pendingAggregationRequests['RequesterAccountId']:
+                    for region in all_regions:
+                        try:
+                            client_acr = account_session.client('config', region_name=region)
+                            response = client_acr.put_aggregation_authorization(
+                                AuthorizedAccountId=tseclog_id,
+                                AuthorizedAwsRegion='eu-west-1'
+                            )
+                        except ClientError as error:
+                            if  error.response['Error']['Code'] != 'AccessDeniedException':
+                                raise error
+                            else:
+                                print(f"{error}{region}")
+                        
+                    
+
+            print(f"\033[2K\033[1GAsassociate AWSConfig linked account to target SECLOG account [{Status.OK.value}]")
+        except botocore.exceptions.ClientError as error:
+                if error.response['Error']['Code'] == 'BadRequestException':
+                    if 'is not associated' in error.response['Error']['Message']:
+                        print(f"\033[2K\033[1GAssociate AWSConfig linked account to target SECLOG account [{Status.NO_ACTION.value}]")
+                    else:
+                        print(f"\033[2K\033[1GAssociate AWSConfig linked account to target SECLOG account [{Status.FAIL.value}]")
+                        print(error.response['Error']['Message'])
+                        print("Exiting...")
+                        sys.exit(1)
+                else:
+                    print(f"\033[2K\033[1GAsssociate AWSConfig linked account to target SECLOG account [{Status.FAIL.value}]")
+                    print(error.response['Error']['Message'])
+                    print("Exiting...")
+                    sys.exit(1)
+
+
+
+def deactivate_guardduty(sseclog_id, session):
+    guardduty = session.client('guardduty')
+    
+    detector_response = guardduty.list_detectors()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `list_detectors` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `IsTruncated`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)